pkg/apis/batch/v1/types.go

```diff
@@ -18,11 +18,55 @@ package v1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/kubernetes/pkg/api/v1"
 )
 
 // +genclient=true
 
+// QueueJob represents the configuration of a QueueJob.
+type QueueJob struct {
```

Comment:
In the design doc, we'd like to have a wrapper object for all API objects. But re-think it today, maybe convert Job, RC, RS into an internal object seems OK.